OverlayRegions should handle OverflowScrollProxyNodes correctly

etiennesegonzac opened the pull request:
#### 211c2ae5026452faa383fbed2f6ef4da2cc44c59
<pre>
OverlayRegions should handle OverflowScrollProxyNodes correctly
<a href="https://bugs.webkit.org/show_bug.cgi?id=294408">https://bugs.webkit.org/show_bug.cgi?id=294408</a>
&lt;<a href="https://rdar.apple.com/141299377">rdar://141299377</a>&gt;

Reviewed by Simon Fraser.

When configuring a ScrollView for OverlayRegions, associate the layers
for any overflow related node (scrolling proxy).

* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(configureScrollViewWithOverlayRegionsIDs):
Call the new `_associateRelatedLayersForOverlayRegions`.
(-[WKWebView _selectOverlayRegionScrollView:]):
(-[WKWebView _resetOverlayRegions]):
(-[WKWebView _updateOverlayRegions:destroyedLayers:]):
Update method signatures for the new `OverlayRegionCandidatesMap`.

* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteScrollingCoordinatorProxyIOS.h:
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteScrollingCoordinatorProxyIOS.mm:
(WebKit::RemoteScrollingCoordinatorProxyIOS::overlayRegionCandidates const):
(WebKit:: const): Deleted.
The `overlayRegionCandidates` method now returns a Map with the associated
layers for overflow related nodes.

* Source/WebKit/UIProcess/ios/WKBaseScrollView.h:
* Source/WebKit/UIProcess/ios/WKBaseScrollView.mm:
(-[WKBaseScrollView _associateRelatedLayersForOverlayRegions:with:]):
Add a stub for the new method in Additions.

* Source/WebKit/UIProcess/API/ios/WKWebViewTestingIOS.mm:
(dumpUIView):
Add an associated layers count to the dump format.
* LayoutTests/overlay-region/full-page-overflow-expected.txt:
* LayoutTests/overlay-region/full-page-overflow-scrolling-expected.txt:
* LayoutTests/overlay-region/snapping-expected.txt:
Update tests expectations with associated layers.
* LayoutTests/overlay-region/overflow-with-proxy-expected.txt: Added.
* LayoutTests/overlay-region/overflow-with-proxy.html: Added.
Add a new test.

Canonical link: <a href="https://commits.webkit.org/296201@main">https://commits.webkit.org/296201@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/86af715c962f7fbced61732b3c25866e7535e26a

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/107679 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27353 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/17768 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/112889 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58215 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28045 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/35876 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81761 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110613 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22227 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97052 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62140 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21661 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15193 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57651 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91599 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15225 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116010 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/34755 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/25615 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90799 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35132 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93302 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90549 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23091 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35465 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13239 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/30510 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34659 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/40215 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34405 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/37766 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36068 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr reviewed (approved):
Can we write some tests?

smfr commented on Source/WebKit/UIProcess/API/ios/WKWebViewTestingIOS.mm:
```diff
@@ -261,6 +261,11 @@ static void dumpUIView(TextStream& ts, UIView *view)
         for (NSString *overlayAsString in overlaysAsStrings.get())
             ts.dumpProperty("overlay region"_s, overlayAsString);
+
+        auto associatedLayers = [(WKBaseScrollView *)view overlayRegionAssociatedLayersForTesting];
```
auto&

etiennesegonzac committed:
OverlayRegions should handle OverflowScrollProxyNodes correctly
https://bugs.webkit.org/show_bug.cgi?id=294408
<rdar://141299377>

Reviewed by Simon Fraser.

When configuring a ScrollView for OverlayRegions, associate the layers
for any overflow related node (scrolling proxy).

* Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm:
(configureScrollViewWithOverlayRegionsIDs):
Call the new `_associateRelatedLayersForOverlayRegions`.
(-[WKWebView _selectOverlayRegionScrollView:]):
(-[WKWebView _resetOverlayRegions]):
(-[WKWebView _updateOverlayRegions:destroyedLayers:]):
Update method signatures for the new `OverlayRegionCandidatesMap`.

* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteScrollingCoordinatorProxyIOS.h:
* Source/WebKit/UIProcess/RemoteLayerTree/ios/RemoteScrollingCoordinatorProxyIOS.mm:
(WebKit::RemoteScrollingCoordinatorProxyIOS::overlayRegionCandidates const):
(WebKit:: const): Deleted.
The `overlayRegionCandidates` method now returns a Map with the associated
layers for overflow related nodes.

* Source/WebKit/UIProcess/ios/WKBaseScrollView.h:
* Source/WebKit/UIProcess/ios/WKBaseScrollView.mm:
(-[WKBaseScrollView _associateRelatedLayersForOverlayRegions:with:]):
Add a stub for the new method in Additions.

* Source/WebKit/UIProcess/API/ios/WKWebViewTestingIOS.mm:
(dumpUIView):
Add an associated layers count to the dump format.
* LayoutTests/overlay-region/full-page-overflow-expected.txt:
* LayoutTests/overlay-region/full-page-overflow-scrolling-expected.txt:
* LayoutTests/overlay-region/snapping-expected.txt:
Update tests expectations with associated layers.
* LayoutTests/overlay-region/overflow-with-proxy-expected.txt: Added.
* LayoutTests/overlay-region/overflow-with-proxy.html: Added.
Add a new test.

Canonical link: https://commits.webkit.org/296201@main
